heron/schedulers/src/java/com/twitter/heron/scheduler/local/LocalScheduler.java

```diff
@@ -261,12 +262,15 @@ public boolean onUpdate(Scheduler.UpdateTopologyRequest request) {
   @Override
   public void addContainers(Set<PackingPlan.ContainerPlan> containers) {
     synchronized (processToContainer) {
+      Set<Integer> containerIds = new HashSet<>();
+      containerIds.addAll(processToContainer.values());
       for (PackingPlan.ContainerPlan container : containers) {
-        if (processToContainer.containsKey(container.getId())) {
+        if (containerIds.contains(container.getId())) {
```

Comment:
Sorry that I was not clear. I meant the `contains` method is `O(# existing containers)`.